src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -139,34 +34,82 @@ def read_slr_data_from_files(slr_data_dir: pathlib.Path) -> gpd.GeoDataFrame:
+    # Get the url where sea level rise files need downloading
+    url = 'https://zenodo.org/records/11398538/export/json'
+
+    # Create dictionary to store the sea level rise dataset
+    slr_nz_dict = {}
+    # Request export information json from Zenodo
+    response = requests.get(url)
+    response.raise_for_status()
+    export_json = response.json()
+    # Get necessary links from json file
+    for file_name, file_info in export_json['files']['entries'].items():
+        # Get file name without extension
+        file_name_without_extension = os.path.splitext(file_name)[0]
+        # Collect sea level rise dataset and storing into dictionary
+        slr_nz_dict[file_name_without_extension] = pd.read_csv(file_info["links"]["content"])
+
+        # Check response for errors
+        response.raise_for_status()
+
+    # Column name used to merge
+    left_column_name = 'Site ID'
+    right_column_name = 'site'
+    # Merge Site Details dataframe and Sea level projections tables WITH VLM
+    slr_nz_merge_vlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_vlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_vlm['add_vlm'] = True
+
+    # Merge Site Details dataframe and Sea level projections tables WITHOUT VLM
+    slr_nz_merge_novlm = pd.merge(
+        slr_nz_dict["NZ_VLM_final_May24"], slr_nz_dict["NZSeaRise_proj_novlm"],
+        left_on=left_column_name, right_on=right_column_name,
+        how='left', validate='1:m',
+        suffixes=('_y', '')
+    )
+    slr_nz_merge_novlm['add_vlm'] = False
+
+    # Concatenate all the dataframes (both WITH VLM and WITHOUT VLM)
+    slr_nz = pd.concat([slr_nz_merge_vlm, slr_nz_merge_novlm], axis=0).reset_index(drop=True)
+
+    # Remove unnamed columns
+    slr_nz_df = slr_nz.loc[:, ~slr_nz.columns.str.contains('^Unnamed')]
+    # Remove site column
+    slr_nz_df = slr_nz_df.drop(columns=['site'])
+    # Rename the columns
+    slr_nz_df = slr_nz_df.rename(columns={
+        left_column_name: 'siteid',
+        'Lon': 'lon',
+        'Lat': 'lat',
+        'Vertical Rate (mm/yr)': 'vertical_rate',
+        'Vertical Rate - BOP corrected (mm/yr)': 'vertical_rate_bop',
+        '1-sigma uncertainty (mm/yr)': 'sigma_uncertainty',
+        'Number of obs': 'number_of_obs',
+        'Quality Factor': 'quality_factor',
+        'Average distance between coastal point and observations': "average_distance",
+        'Confidence': 'confidence_level',
+        '0.17': 'p17',
+        '0.5': 'p50',
+        '0.83': 'p83',
+        'SSP': 'ssp'
+    })
+
+    # Remove '_confidence' in confidence_level column
+    slr_nz_df['confidence_level'] = slr_nz_df['confidence_level'].str.split('_').str[0]
+    # Merge SSP and scenario into one column 'ssp_scenario'
+    slr_nz_df['ssp_scenario'] = slr_nz_df['ssp'].astype(str) + '-' + slr_nz_df['scenario'].astype(str)
```

Comment:
I think we should keep it as similar to the original takiwa data source as possible, so do not combine ssp and scenario

Reply:
@Martin20494 The old raw data used to have them combined into one column, but the new raw data has them separated out. We always aim to store the original raw data in the database.